account_payment_base_oca/models/account_payment_method_line.py

```diff
@@ -0,0 +1,195 @@
+        required=True,
+        compute="_compute_company_id",  # for smooth post-install
+        store=True,
+        precompute=True,
+        readonly=False,
+    )
+    # END inherit of native fields
+    # Here is the strategy to support bank_account_link = variable
+    # without breaking the native behavior
+    # company_id is a related of journal_id.company_id
+    # When bank_account_link = 'fixed' => we use journal_id
+    # When bank_account_link = 'variable':
+    # - journal_id is considered as the default journal
+    # - alternative_journal_ids are additional journals that can be used
+    filter_journal_ids = fields.Many2many(
+        "account.journal", compute="_compute_filter_journal_ids"
+    )
+    bank_account_link = fields.Selection(
+        [("fixed", "Fixed"), ("variable", "Variable")],
+        string="Link to Bank Account",
+        required=True,
+        default="fixed",
+        help="For payment modes that are always attached to the same bank "
+        "account of your company (such as wire transfer from customers or "
+        "SEPA direct debit from suppliers), select "
+        "'Fixed'. For payment modes that are not always attached to the same "
+        "bank account (such as SEPA Direct debit for customers, wire transfer "
+        "to suppliers), you should select 'Variable', which means that you "
+        "will select the bank account on the payment order. If your company "
+        "only has one bank account, you should always select 'Fixed'.",
+    )
+    # I need to explicitly define the table name
+    # because I have 2 M2M fields pointing to account.journal
+    alternative_journal_ids = fields.Many2many(
+        comodel_name="account.journal",
+        relation="account_payment_method_line_alternative_journal_rel",
+        column1="method_line_id",
+        column2="journal_id",
+        string="Alternative Bank Journals",
+        domain="[('id', '!=', journal_id), ('id', 'in', filter_journal_ids)]",
+        check_company=True,
+        compute="_compute_alternative_journal_ids",
+        store=True,
+        readonly=False,
+        precompute=True,
+    )
+    # active is default=False on purpose !
+    # When an account.payment.method with mode='multi' is created, Odoo
+    # generates an account.payment.method.line for each bank journal and there is no
+    # prepare method to inherit the values of the generated method lines.
+    # With default=False, Odoo will auto-generate inactive account.payment.method.line
+    # and we will only enable the one we want to use. If a payment order has a method
+    # line with bank_account_link='variable' and an alternative journal is selected,
+    # Odoo will not use the method line of the payment order to generate
+    # the account.payment but it will select the (inactive) method line
+    # linked to the chosen journal
+    # TODO default=False causes problems in tests of the account module
+    active = fields.Boolean(default=False)
```

Comment:
Would it make sense to set active=False on create only when bank_account_link is variable?